lib/server.js

```diff
@@ -100,6 +112,49 @@ Server.prototype._initializeOptions = function(options) {
   this.wsdl.options.attributesKey = options.attributesKey || 'attributes';
 };
 
+Server.prototype._processRequestXml = function(req, res, xml){
+  var self = this;
+  var result;
+  var error;
+  try {
+    if (typeof self.log === 'function') {
+      self.log("received", xml);
+    }
+    self._process(xml, req, function(result, statusCode) {
+      if(statusCode) {
+        res.statusCode = statusCode;
+      }
+      res.write(result);
+      res.end();
+      if (typeof self.log === 'function') {
+        self.log("replied", result);
+      }
+    });
+  }
+  catch (err) {
+    if (err.Fault !== undefined) {
+      return self._sendError(err.Fault, function(result, statusCode) {
+        if(statusCode) {
+          res.statusCode = statusCode || 500;
```

Comment:
either we don't need the `if` around the assignment or the `|| 500` as we won't hit it if `statusCode` does not exist.